tcms/core/ajax.py

```diff
@@ -120,14 +119,14 @@ def env_values(self):
             )
 
         def tags(self):
-            query = strip_parameters(request, self.internal_parameters)
-            tags = TestTag.objects
+            query = strip_parameters(request.GET, self.internal_parameters)
+            tags = TestCaseTag.objects
```

Comment:
This is the same reversed child -> parent (or parent -> child) relationship that we talked about in another PR. In this case the relationship goes from TestCaseTag (child) to TestTag (parent). I think it is more natural and more understandable to always use the child -> parent side of the relationship, especially when it is very easy to do so.

The backwards relationship is possible in Django ORM but IMO it is less clear and TBH in many cases it doesn't bring any other benefits. Behind the scenes the parent -> child ORM queries are translated to inner joins which I think are also less efficient.